semeio/jobs/rft/utility.py

```diff
@@ -68,21 +68,21 @@ def load_and_parse_well_time_file(filename):
 
         try:
             report_step = int(tokens[4])
-        except ValueError:
+        except ValueError as err:
```

Comment:
What do you lose?

Reply:
The `argparse.ArgumentTypeError` (or at least the intent of it) is to have a nicely formed error message to the user, without the additional traceback. The `from err` will append said traceback never the less. And instead of just getting the `Unable to convert <something> to int`, you will get the entire traceback dumped to terminal, with nothing more then the same problem..

Reply:
Except you dont, you catch the error, so it does not matter, this is the output to the user:
```
tests/jobs/rft/test_gendata_rft.py usage: script_name [-h] -e ECLBASE -w WELL_AND_TIME_FILE -t TRAJECTORY_PATH
                   [-z ZONEMAP] [-c CSVFILE] [-o OUTPUTDIRECTORY]
                   [--log-level LOG_LEVEL]
script_name: error: argument -w/--well_and_time_file: Line 8 in well_and_time_file well_and_time.txt not on proper format: Unable to convert not_a_number to int
```

Reply:
That's not what I got when I tested it - hang on

Reply:
So, I don't get it pretty much.. :P 

I believe I only tested the specific function itself, and not while I was used in argparse? maybe? 

Anywho, why are we adding something that does nothing to the output? Or, more importantly, why doesn't it do affect the stderror? Is it because the `ArgParse` package catches all `ArgumentTypeError`s and only forwards the output? Are we kind of dependent of the behaviour of the `ArgumentTypeError` here? And could it be that they will update the behaviour when using `from err`?

Reply:
> Anywho, why are we adding something that does nothing to the output?

Because it improves the traceback, so it should make it easier for the developer, and the user should have the same experience. `argparse` is catching the error, not the full traceback, and probably printing `str(err)` on the error.

Reply:
>Anywho, why are we adding something that does nothing to the output?

That was a really poorly formulated question.. 

>Because it improves the traceback, so it should make it easier for the developer

I'm still not completely convinced.. I suppose I'm arguing the question of "Should absolutely every re-raise use `from err` unconditionally"?

Reply:
Did a bit more testing..

Changed to raising an `Exception` just to get it to throw to terminal. With original implementation:

```
Traceback (most recent call last):
  File "/Users/LPHA/venvs/ert_env/lib/python3.8/site-packages/semeio/jobs/rft/utility.py", line 70, in load_and_parse_well_time_file
    report_step = int(tokens[4])
ValueError: invalid literal for int() with base 10: 'h'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/LPHA/venvs/ert_env/bin/gendata_rft", line 8, in <module>
    sys.exit(main_entry_point())
  File "/Users/LPHA/venvs/ert_env/lib/python3.8/site-packages/semeio/jobs/scripts/gendata_rft.py", line 173, in main_entry_point
    options = arg_parser.parse_args()
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1768, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1800, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 2006, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1946, in consume_optional
    take_action(action, args, option_string)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1858, in take_action
    argument_values = self._get_values(action, argument_strings)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 2389, in _get_values
    value = self._get_value(action, arg_string)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 2422, in _get_value
    result = type_func(arg_string)
  File "/Users/LPHA/venvs/ert_env/lib/python3.8/site-packages/semeio/jobs/rft/utility.py", line 73, in load_and_parse_well_time_file
    raise Exception(
Exception: Line 2 in well_and_time_file gendata_rft_input_files/well_and_time.txt not on proper format: Unable to convert h to int
```

Using `from err`:

```
Traceback (most recent call last):
  File "/Users/LPHA/venvs/ert_env/lib/python3.8/site-packages/semeio/jobs/rft/utility.py", line 70, in load_and_parse_well_time_file
    report_step = int(tokens[4])
ValueError: invalid literal for int() with base 10: 'h'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/LPHA/venvs/ert_env/bin/gendata_rft", line 8, in <module>
    sys.exit(main_entry_point())
  File "/Users/LPHA/venvs/ert_env/lib/python3.8/site-packages/semeio/jobs/scripts/gendata_rft.py", line 173, in main_entry_point
    options = arg_parser.parse_args()
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1768, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1800, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 2006, in _parse_known_args
    start_index = consume_optional(start_index)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1946, in consume_optional
    take_action(action, args, option_string)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 1858, in take_action
    argument_values = self._get_values(action, argument_strings)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 2389, in _get_values
    value = self._get_value(action, arg_string)
  File "/usr/local/Cellar/python@3.8/3.8.6_1/Frameworks/Python.framework/Versions/3.8/lib/python3.8/argparse.py", line 2422, in _get_value
    result = type_func(arg_string)
  File "/Users/LPHA/venvs/ert_env/lib/python3.8/site-packages/semeio/jobs/rft/utility.py", line 73, in load_and_parse_well_time_file
    raise Exception(
Exception: Line 2 in well_and_time_file gendata_rft_input_files/well_and_time.txt not on proper format: Unable to convert h to int
```

I might have misunderstood how it works then, but the only difference is that there's a rephrasing on the reference of the initial error? From 

`The above exception was the direct cause of the following exception:`

to

`During handling of the above exception, another exception occurred:`

?

But then the traceback hasn't changed?